load-test-framework/src/main/java/com/google/pubsub/clients/common/LoadTestRunner.java

```diff
@@ -0,0 +1,134 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+package com.google.pubsub.clients.common;
+
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import com.google.common.util.concurrent.RateLimiter;
+import com.google.common.util.concurrent.SettableFuture;
+import com.google.pubsub.flic.common.LoadtestGrpc;
+import com.google.pubsub.flic.common.LoadtestProto.CheckRequest;
+import com.google.pubsub.flic.common.LoadtestProto.CheckResponse;
+import com.google.pubsub.flic.common.LoadtestProto.StartRequest;
+import com.google.pubsub.flic.common.LoadtestProto.StartResponse;
+import io.grpc.Server;
+import io.grpc.ServerBuilder;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.charset.Charset;
+import java.util.Arrays;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.function.Function;
+
+/**
+ * Starts a server to get the start request, then starts the client runner.
+ */
+public class LoadTestRunner {
+  private static final Logger log = LoggerFactory.getLogger(LoadTestRunner.class);
+  private static Server server;
+  private static Task client;
+  private static AtomicBoolean finished = new AtomicBoolean(false);
+  private static SettableFuture<Void> finishedFuture = SettableFuture.create();
+
+  public static void run(Function<StartRequest, Task> loadFunction) throws Exception {
+    SettableFuture<StartRequest> requestFuture = SettableFuture.create();
+    server = ServerBuilder.forPort(5000)
+        .addService(new LoadtestGrpc.LoadtestImplBase() {
+          @Override
+          public void start(StartRequest request, StreamObserver<StartResponse> responseObserver) {
+            if (requestFuture.isDone()) {
+              responseObserver.onError(new Exception("Start should only be called once, ignoring this request."));
+              return;
+            }
+            requestFuture.set(request);
+            responseObserver.onNext(StartResponse.getDefaultInstance());
```

Comment:
Alright, so I guess there are no "synchronous" errors that we can return? Like, if input validation fails?